torchaudio/compliance/kaldi.py

```diff
@@ -512,3 +514,172 @@ def fbank(
+    """ Based on LinearResample::GetNumOutputSamples
+    Inputs:
+        input_num_samp (int): the number of samples in the input
+        samp_rate_in (float): the original frequency of the signal
+        samp_rate_out (float): the desired frequency
+
+    Outputs:
+        int: the number of output samples
+    """
+    # For exact computation, we measure time in "ticks" of 1.0 / tick_freq,
+    # where tick_freq is the least common multiple of samp_rate_in and
+    # samp_rate_out.
+    tick_freq = _lcm(samp_rate_in, samp_rate_out)
+    ticks_per_input_period = tick_freq // samp_rate_in
+
+    # work out the number of ticks in the time interval
+    # [ 0, input_num_samp/samp_rate_in ).
+    interval_length_in_ticks = input_num_samp * ticks_per_input_period
+    if interval_length_in_ticks <= 0:
+        return 0
+    ticks_per_output_period = tick_freq // samp_rate_out
+    # Get the last output-sample in the closed interval, i.e. replacing [ ) with
+    # [ ].  Note: integer division rounds down.  See
+    # http://en.wikipedia.org/wiki/Interval_(mathematics) for an explanation of
+    # the notation.
+    last_output_samp = interval_length_in_ticks // ticks_per_output_period
+    # We need the last output-sample in the open interval, so if it takes us to
+    # the end of the interval exactly, subtract one.
+    if last_output_samp * ticks_per_output_period == interval_length_in_ticks:
+        last_output_samp -= 1
+    # First output-sample index is zero, so the number of output samples
+    # is the last output-sample plus one.
+    num_output_samp = last_output_samp + 1
+    return num_output_samp
+
+
+def resample_waveform(wave, orig_freq, new_freq):
+    """Resamples the wave at the new frequency. This matches Kaldi's OfflineFeatureTpl ResampleWaveform
+    which uses a LinearResample (resample a signal at linearly spaced intervals to upsample/downsample
+    a signal).
+
+    Inputs:
+        wave (Tensor): the input signal of size (c, n)
+        orig_freq (float): the original frequency of the signal
+        new_freq (float): the desired frequency
+
+    Outputs:
+        Tensor: the signal at the new frequency
+    """
+    assert orig_freq > 0.0 and new_freq > 0.0
+
+    min_freq = min(orig_freq, new_freq)
+    lowpass_cutoff = 0.99 * 0.5 * min_freq
+    lowpass_filter_width = 6
+
+    assert lowpass_cutoff * 2 <= min_freq
+
+    base_freq = math.gcd(int(orig_freq), int(new_freq))
+    input_samples_in_unit = int(orig_freq) // base_freq
+    output_samples_in_unit = int(new_freq) // base_freq
+
+    window_width = lowpass_filter_width / (2.0 * lowpass_cutoff)
+    first_indices, weights = get_LR_indices_and_weights(orig_freq, new_freq, output_samples_in_unit,
+                                                        window_width, lowpass_cutoff, lowpass_filter_width)
+
+    assert first_indices.dim() == 1
+    # TODO figure a better way to do this. conv1d reaches every element i*stride + padding
```

Comment:
What is not great about this current implementation?

Reply:
I think it would be nice to not have a for loop. Currently it applies each filter separately but ideally, torch.conv1d allowed different paddings per filter e.g. ``padding=(filter1_padding, filter2_padding, ...)``. Something to look at in the future

Reply:
Ok, you can mention the for loop in the comment.